src/kernel/src/fs/mount.rs

```diff
@@ -153,8 +154,227 @@ bitflags! {
     }
 }
 
+pub(super) struct MountOpts(HashMap<&'static str, MountOpt>);
+
+impl MountOpts {
+    pub fn new() -> Self {
+        Self(HashMap::new())
+    }
+
+    pub fn insert(&mut self, k: &'static str, v: impl Into<MountOpt>) {
+        self.0.insert(k.into(), v.into());
+    }
+
+    pub fn remove(&mut self, k: &'static str) -> Option<MountOpt> {
+        self.0.remove(k)
+    }
+
+    pub fn retain(&mut self, mut f: impl FnMut(&str, &mut MountOpt) -> bool) {
+        self.0.retain(|k, v| f(*k, v));
+    }
+}
+
+pub(super) enum MountOpt {
+    Bool(bool),
+    Int(i32),
+    Usize(usize),
+    Str(Box<str>),
+    VPath(VPathBuf),
+    Path(PathBuf),
+    Param(Arc<Param>),
+    Gid(Gid),
+    Uid(Uid),
+    Mode(Mode),
+}
+
+impl MountOpt {
+    pub fn as_bool(&self) -> Option<bool> {
+        match self {
+            Self::Bool(v) => Some(*v),
+            _ => None,
+        }
+    }
+}
+
+impl From<bool> for MountOpt {
+    fn from(v: bool) -> Self {
+        Self::Bool(v)
+    }
+}
+
+impl From<i32> for MountOpt {
+    fn from(v: i32) -> Self {
+        Self::Int(v)
+    }
+}
+
+impl From<usize> for MountOpt {
+    fn from(v: usize) -> Self {
+        Self::Usize(v)
+    }
+}
+
+impl From<&str> for MountOpt {
+    fn from(v: &str) -> Self {
+        Self::Str(v.into())
+    }
+}
+
+impl From<String> for MountOpt {
+    fn from(v: String) -> Self {
+        Self::Str(v.into_boxed_str())
+    }
+}
+
+impl From<VPathBuf> for MountOpt {
+    fn from(v: VPathBuf) -> Self {
+        Self::VPath(v)
+    }
+}
+
+impl From<PathBuf> for MountOpt {
+    fn from(v: PathBuf) -> Self {
+        Self::Path(v)
+    }
+}
+
+impl From<Arc<Param>> for MountOpt {
+    fn from(v: Arc<Param>) -> Self {
+        Self::Param(v)
+    }
+}
+
+impl From<Gid> for MountOpt {
+    fn from(v: Gid) -> Self {
+        Self::Gid(v)
+    }
+}
+
+impl From<Uid> for MountOpt {
+    fn from(v: Uid) -> Self {
+        Self::Uid(v)
+    }
+}
+
+impl From<Mode> for MountOpt {
+    fn from(v: Mode) -> Self {
+        Self::Mode(v)
+    }
+}
```

Comment:
These should not be here otherwise this file will be polluted with all of types in the kernel.

Reply:
So... you want me to move them to where every type is defined?